src/engine/guidance/post_processing.cpp

```diff
@@ -815,6 +815,20 @@ std::vector<RouteStep> collapseTurns(std::vector<RouteStep> steps)
                     invalidateStep(steps[step_index]);
                 }
             }
+            else if (step_index + 2 < steps.size() &&
+                     current_step.maneuver.instruction.type == TurnType::NewName &&
+                     steps[step_index + 1].maneuver.instruction.type == TurnType::NewName &&
```

Comment:
can you name `steps[step_index + 1]`? I think it is heard to see the realtion between `current_step` and `one_step_back` with this one.